macros/fivetran/base/fivetran_fb_url_tag.sql

```diff
@@ -0,0 +1,81 @@
+{% macro fivetran_fb_url_tag() %}
+
+    {{adapter_macro('facebook_ads.fivetran_fb_url_tag')}}
+
+{% endmacro %}
+
+{% macro default__fivetran_fb_url_tag() %}
+
+with base as (
+    
+select distinct
+    type,
+    key,
+    creative_id,
+    value
+
+from
+    {{ var('url_tag_table') }}
+  
+),
+
+pivot as (
+    
+    select 
+
+        creative_id,
+    
+        {% for key, value in {
+        "affcode": "url_host",
+        "utm_source": "utm_source",
+        "utm_medium": "utm_medium",
+        "utm_campaign": "utm_campaign",
+        "utm_content": "utm_content",
+        "utm_term": "utm_term"
+        }.items()%}
+
+            first_value(
+                case 
+                    when key = '{{key}}' then value
+                    else null  
+                end
+
+                {% if target.type == 'snowflake' %}
```

Comment:
we have adapter macros to prevent you using if/then constructions like this :) can talk about this more but we should never write this particular check; it creates long-term maintenance headaches. instead, write an adapter macro that accomplishes the same thing.

Reply:
changed to self-contained adapter macro